NoClassDefFoundError: org/apache/commons/lang/NotImplementedException when opening a DSS database in the Ensemble Viewer

After integrating the Ensemble Viewer into HMS, opening a DSS database file crashes the AWT event thread with a NoClassDefFoundError for org.apache.commons.lang.NotImplementedException. The viewer window opens, but selecting a DSS file fails.                  
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             
Caused by: java.lang.ClassNotFoundException: org.apache.commons.lang.NotImplementedException                                                                                                                                                                                                                     
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    
hec.dss.ensemble.DssDatabase (module ensemble-dss) referenced org.apache.commons.lang.NotImplementedException — a class from Apache Commons Lang 2.x (commons-lang:commons-lang, the legacy org.apache.commons.lang package, not commons-lang3). It was used only in four unimplemented stub methods.    

That class was never declared as a direct dependency; it arrived transitively through hec-monolith (declared api in ensemble-dss). When HMS consumes ensemble-view it excludes hec-monolith (HMS supplies its own), which removed the transitive commons-lang 2.x from the runtime classpath. The reference therefore resolved at build time (precompiled jar) but failed to link at runtime → NoClassDefFoundError.                                                                                                                                                                                                         
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   
Removed the dependency on Commons Lang 2.x in the library by replacing NotImplementedException with the JDK-native java.lang.UnsupportedOperationException. This is behavior-preserving: org.apache.commons.lang.NotImplementedException extends UnsupportedOperationException and is unchecked, so the swap keeps the same "not implemented" semantics with no new dependency.                                                                                                                                                                                                                                               
                                                                                                                                                                                                                                                                                                                   File: ensemble-dss/src/main/java/hec/dss/ensemble/DssDatabase.java                                                                                                                                                                                                                                               
  - Removed import org.apache.commons.lang.NotImplementedException;                                                                                                                                                                                                                                                
  - Replaced all usages (throw new NotImplementedException() and the matching throws clauses) in getMetricCollectionTimeSeries(...) (x2) and getMetricStatistics(...) (x2) with UnsupportedOperationException.                                                                                                     
                                                                                                                                                                                                                                                                                                                  A repo-wide audit confirmed commons-lang 2.x had no other usages in ensemble-dss or ensemble-view (the only remaining reference was a code comment in FIRO_TSEnsembles/.../Statistics.java), so the library no longer depends on Commons Lang 2.x at all and no downstream build.gradle change is required.

--- a/ensemble-dss/src/main/java/hec/dss/ensemble/DssDatabase.java
+++ b/ensemble-dss/src/main/java/hec/dss/ensemble/DssDatabase.java
@@ -15,7 +15,6 @@ import hec.io.TimeSeriesCollectionContainer;
 import hec.io.TimeSeriesContainer;
 import hec.metrics.MetricCollection;
 import hec.metrics.MetricCollectionTimeSeries;
-import org.apache.commons.lang.NotImplementedException;
 
 import java.time.Duration;
 import java.time.LocalDateTime;
@@ -374,23 +373,23 @@ public class DssDatabase implements EnsembleDatabase,MetricDatabase {
     }*/
 
     @Override
-    public List<hec.metrics.MetricCollectionTimeSeries> getMetricCollectionTimeSeries(hec.RecordIdentifier timeseriesID) throws NotImplementedException{
-        throw new NotImplementedException();
+    public List<hec.metrics.MetricCollectionTimeSeries> getMetricCollectionTimeSeries(hec.RecordIdentifier timeseriesID) throws UnsupportedOperationException{
+        throw new UnsupportedOperationException();
     }
 
     @Override
-    public hec.metrics.MetricCollectionTimeSeries getMetricCollectionTimeSeries(hec.RecordIdentifier timeseriesID, String statistic) throws NotImplementedException {
-        throw new NotImplementedException();
+    public hec.metrics.MetricCollectionTimeSeries getMetricCollectionTimeSeries(hec.RecordIdentifier timeseriesID, String statistic) throws UnsupportedOperationException {
+        throw new UnsupportedOperationException();
     }
 
     @Override
-    public Map<RecordIdentifier, List<String>> getMetricStatistics() throws NotImplementedException {
-        throw new NotImplementedException();
+    public Map<RecordIdentifier, List<String>> getMetricStatistics() throws UnsupportedOperationException {
+        throw new UnsupportedOperationException();
     }
 
     @Override
-    public List<String> getMetricStatistics(hec.RecordIdentifier timeseriesID) throws NotImplementedException {
-        throw new NotImplementedException();
+    public List<String> getMetricStatistics(hec.RecordIdentifier timeseriesID) throws UnsupportedOperationException {
+        throw new UnsupportedOperationException();
     }
 
     private float[][] getMetricValues(List<DSSPathname> paths) {

